src/luarocks/init.lua

```diff
@@ -1295,4 +1228,9 @@ function luarocks.check_missing_dependencies(name, version, tree)
    return nil, "No .rockspec or .rock can be found."
 end
 
-return luarocks
\ No newline at end of file
+for name, func in pairs(doc) do
+   luarocks[name] = func
+end
```

Comment:
I don't like this "mindless" copying of contents:

-  going forward, when more parts get factored out, this API becomes a long list of those copy-loops, which makes the code base harder to understand
- doesn't allow for adding `ldoc` inline comments for documentation generation for the api.
- future accidental additions to modules get automatically exported this way
- above all: I prefer explicitness

Reply:
In the last update I exported the API functions in the following way:
```
luarocks = {
   doc = doc.doc,
   homepage = doc.homepage
}
```
Is this what you meant through explicit exporting?